### DISCUSSION TITLE
Trouble updating on Windows

### DISCUSSION THREAD

**Discussion by porentief49** [JanDeDobbeleer/oh-my-posh#2171] [Troubleshoot] [answered]
Hi!

I've been running "a" version of this for a while without any troubles and without really paying attention to any upgrades. Recently, I experienced the "invalid memory address or nil pointer dereference" error to happen occassionally. I learned that some fixes have been implemented and wanted to update. I have to add I'm in the first half of the learning curve on Terminal / Powershell / command line stuff.

With `Get-Module` I found out that I have version 7.59.1 on one machine and 7.54.0 on the other (which is a bit strange as I set up the second one much later than the first).

I tried to `winget upgrade JanDeDobbeleer.OhMyPosh`, but that told me that "No installed package found matching input criteria." Next, I tried `winget install JanDeDobbeleer.OhMyPosh`, which successfully downloaded and installed "https://github.com/JanDeDobbeleer/oh-my-posh/releases/download/v7.72.5/install-amd64.exe". However, I don't have an AMD64 processor (Intel Core I7 and I9 instead), and the `Get-Module` command still reports the old versions.

I must be doing something wrong, but I can't figure out what. Any ideas?

**Comment by JanDeDobbeleer** [comment, marked as answer]
Thanks for reaching out. The fact that `upgrade` doesn't work is because it wasn't previously installed using winget. Installing with winget does not add a module to PowerShell, it installs the oh-my-posh binary and themes on your machine. You can check which version is added to your PATH using `oh-my-posh version`. As the module is now no longer offered, you can uninstall the module(s) on both machines (see [here](https://ohmyposh.dev/docs/migrating)) and rely solely on the executable (no functionality loss whatsoever). amd64 works for intel as well, that's the 64-bit executable.

**Reply by porentief49** [reply]
Wow - problem solved, thanks a lot for the prompt (!) feedback ...
